File: backend/src/main/java/de/neuefische/backend/model/Organization.java
package de.neuefische.backend.model;


import jakarta.validation.Valid;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

public record Organization(String id, String name, String homepage, String email, String address,
                           List<@Valid Review> reviews, double averageRating) {
    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;
        if (o == null || getClass() != o.getClass())
            return false;
        Organization that = (Organization) o;
        return Objects.equals(name, that.name);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(name);
    }
}

File: frontend/src/types/Organization.ts
import {Review} from "./Review.ts";
import {Course} from "./Course.ts";

export type Organization =
    {
        id: string;
        name: string;
        homepage: string;
        email: string;
        address: string;
        reviews: Review[];
        averageRating: number;
        courses: Course[]
    }